Autopilot/Src/MPXV7002DP.cpp

```diff
@@ -0,0 +1,65 @@
+#include "airspeed.hpp"
+#include "adc.h"
+#include <stddef.h>
+#include <cmath>
+
+#define VS 5
+#define DENSITY_AIR 1.204
+#define ADC_RESOLUTION 4096
+
+
+MPXV7002DP::MPXV7002DP() {
+    //constructor
+    MX_ADC2_Init();
+}
+
+MPXV7002DP* MPXV7002DP::s_Instance = NULL;
+
+MPXV7002DP* MPXV7002DP::GetInstance() {
+    if(!s_Instance) {
+        s_Instance = new MPXV7002DP;
+    }
+    return s_Instance;
+}
+
+void MPXV7002DP::GetResult(airspeedData_t &Data) {
+        Data.airspeed = airspeed;
+        Data.isDataNew = dataNew;
+        dataNew = false;
+        Data.sensorStatus = sensorStatus;
+        Data.utcTime = timeOfLastResult;
+}
+
+void MPXV7002DP::Begin_Measuring() {
+        airspeed = readFromMPXV7002DP();
+        timeOfLastResult = getCurrentTime();
+        dataNew = true;
+        sensorStatus = 0;
+}
+
+void MPXV7002DP::Init() {
+
+    //do I need to init the adc?
```

Comment:
all initialization is done by the MX_adc_init function, no need to do anything else